matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TimelineChunk.kt

```diff
@@ -430,6 +430,31 @@ internal class TimelineChunk(private val chunkEntity: ChunkEntity,
     private fun handleDatabaseChangeSet(results: RealmResults<TimelineEventEntity>, changeSet: OrderedCollectionChangeSet) {
         val insertions = changeSet.insertionRanges
         for (range in insertions) {
+            // Check if the insertion's displayIndices match our expectations - or skip this insertion.
+            // Inconsistencies (missing messages) can happen otherwise if we get insertions before having loaded all timeline events of the chunk.
+            if (builtEvents.isNotEmpty()) {
+                // Check consistency to item before insertions
+                if (range.startIndex > 0) {
+                    val firstInsertion = results[range.startIndex]!!
+                    val lastBeforeInsertion = builtEvents[range.startIndex - 1]
```

Comment:
If there is no built event yet, this could crash

Reply:
There is `builtEvents.isNotEmpty()` in line 435

Reply:
Oh yeah; I'm still wandering if we should try to fix the underneath issue here instead of doing some weird checks :/